core/psh/runfile/runfile.c

```diff
@@ -26,10 +26,14 @@
 int psh_runfile(int argc, char **argv)
 {
 	pid_t pid;
+	int err;
 
 	pid = vfork();
 	if (pid > 0) {
-		waitpid(pid, NULL, 0);
+		waitpid(pid, &err, 0);
+		if (err < 0) {
+			fprintf(stderr, "psh: %s execution halted with error code %d\n", argv[0], err);
+		}
```

Comment:
I am not sure if we want to printf any comment about child process exit code. Also, only exit codes in range from 0 to 255 (so non-negative) are portably valid

Reply:
Dropped as proposed. Thanks!

Reply:
please note that this is not exit code but status of the child - You should use relevant macros to access the information needed (don't try to understand the value by Yourself) - like `WEXITSTATUS`, `WIFSIGNALED` etc.